runtime/gc_glue_java/HeapWalkerDelegate.cpp

```diff
@@ -58,7 +58,9 @@ void
 MM_HeapWalkerDelegate::doContinuationObject(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, MM_HeapWalkerSlotFunc function, void *userData)
 {
 	J9VMThread *currentThread = (J9VMThread *)env->getLanguageVMThread();
-	if (VM_VMHelpers::needScanStacksForContinuation(currentThread, objectPtr)) {
+	/* In sliding compaction we must fix slots exactly once. Since we will fixup stack slots of mounted Virtual threads later during root fixup,
```

Comment:
As a minimum, the comment should be adjusted, since HeapWalker is a tool that can be used outside of sliding compaction. 

But more importantly, why is this fix needed for heap walker!? Heap walker does not walk over roots (as it does in compact fixup). 
Could it walk over the same object twice via Remember Set walk. But if so, that has no relevance with thread being mounted or not....

Am I missing something else here?

Reply:
yes, you are right, we don't need scan only once protection here.